Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftJoinSessionActivity.cpp

```diff
@@ -44,14 +44,24 @@ namespace AWSGameLift
         {
             Multiplayer::SessionConnectionConfig sessionConnectionConfig;
             auto createPlayerSessionResult = createPlayerSessionOutcome.GetResult();
-            // TODO: AWSNativeSDK needs to be updated to support this attribute, and it is a must have for TLS certificate enabled fleet
-            //sessionConnectionConfig.m_dnsName = createPlayerSessionResult.GetPlayerSession().GetDnsName().c_str();
-            sessionConnectionConfig.m_ipAddress = createPlayerSessionResult.GetPlayerSession().GetIpAddress().c_str();
+            AZStd::string_view dnsName = createPlayerSessionResult.GetPlayerSession().GetDnsName().c_str();
+            AZStd::string_view ipAddress = createPlayerSessionResult.GetPlayerSession().GetIpAddress().c_str();
+            // When connecting to a game session that is running on a TLS-enabled fleet, you must use the DNS name, not the IP address.
+            // https://docs.aws.amazon.com/gamelift/latest/apireference/API_PlayerSession.html
+            if (dnsName.ends_with(".amazongamelift.com"))
```

Comment:
If the dns name is provided, why not always use the DNS name and then fallback to ipaddress? This will also break in other AWS partitions. For example, TLS fleets in china are probably not rooted in amazongamelift.com

Reply:
First, please check the link I provided. dns name is only required when tls enabled, and the format is given by gamelift.
Second, I don't follow your question, and I think you just answer yourself. If you do have concern about partition, then for non tls use case, we should use ip address which is much safer.

Reply:
Update: Double checked china aws doc, seems like there is no TLS certificate related information, so I guess it is not supported in china region?